### PR TITLE
Add non-intrusive editor config with painful shortterm lint consequences

### LINKED ISSUE #30: Add an editor config
I'll be honest, whenever I make one, it never does anything. But it seems important if properly done.

### PR DESCRIPTION
**Pull request by bencooper222**
This will be disruptive short term but enforcing LF is really necessary. The rest is just gravy. 
Closes #30

### CODE REVIEW

File: src/client/routes/profile/Profile.tsx [hunk 1]
```diff
@@ -54,15 +54,15 @@ export const Profile: React.FunctionComponent<{}> = (): JSX.Element => {
 				draft[PROFILE] = { ...data.getUserByEmail };
 			});
 		}
-	}, [data]);
+	}, [data, setFormData]);
```
Comment (cktang88) [inline src/client/routes/profile/Profile.tsx]: what is this change?

File: src/client/routes/application/Application.tsx [hunk 2]
```diff
@@ -106,7 +106,7 @@ export const Application: FunctionComponent<{}> = (): JSX.Element => {
 		return () => {
 			if (setActionButton) setActionButton(undefined);
 		};
-	}, []);
+	}, [setActionButton]);
```
Comment (cktang88) [inline src/client/routes/application/Application.tsx]: ?

File: src/client/routes/profile/Profile.tsx [hunk 3]
```diff
@@ -54,15 +54,15 @@ export const Profile: React.FunctionComponent<{}> = (): JSX.Element => {
...
 
 	useEffect((): (() => void) => {
 		if (setActionButton) setActionButton(<HeaderButton text="Submit" onClick={submit} />);
 
 		return () => {
 			if (setActionButton) setActionButton(undefined);
 		};
-	}, [formData]);
+	}, [formData, setActionButton, submit]);
```
Comment (cktang88) [inline src/client/routes/profile/Profile.tsx]: ?

**Comment by cktang88** [comment]
This PR title is misleading given how much behavior is changed.

**Comment by cktang88** [comment]
I'd suggest breaking this up into several PRs

**Comment by bencooper222** [comment]
Um I just ran the linter and didn't add this stuff... @leonm1 any idea what's going on?

File: src/client/routes/profile/Profile.tsx (hunk 1 above)
Comment (leonm1) [reply src/client/routes/profile/Profile.tsx]: This is a linter change

**Comment by leonm1** [comment]
These are linter fixes because of react hooks checks. They are fine, only merging because of the `lint-fix` applied to the codebase. We should set up husky and CI to catch these types of things

**Pull request merged**